cmd/manager/main.go

```diff
@@ -107,13 +107,17 @@ var (
 				IngressSyncMode: kube.IngressStrict,
 				IngressClass:    flags.ingressClass,
 			}
-			if flags.defaultIngressController {
-				controllerConfig.IngressSyncMode = kube.IngressDefault
-			}
 			controller := kube.NewController(cmd.Client, controllerConfig)
-			w, err := envoy.NewIngressWatcher(controller, controller,
-				&model.IstioRegistry{ConfigRegistry: controller},
-				cmd.Client, &flags.proxy, flags.ingressSecret, cmd.RootFlags.Namespace)
+			context := &envoy.IngressContext{
```

Comment:
I think `IngressConfig` is more appropriate here - proxy context is used to differentiate  multiple envoy instances, ingress proxies seem to act the same way. We can make a more general proxy context later that includes "ingress role" for the proxy, if you want.